React/Next/Development proxy.md

```diff
@@ -0,0 +1,64 @@
+## Motivation
+
+When developing applications, sometimes you want to test your local changes against data from a different environment. For example: creating a production hotfix requires you to locally connect to the production API. Maybe even testing the solution on a different device e.g. a [phone connected to your machine](https://developers.google.com/web/tools/chrome-devtools/remote-debugging). That API could have http-only cookies that cannot be manipulated on the client and require a proxy that can intercept a request and do some modifications on it.
+
+## The issue
+
+NextJS already supports/recommends ways to proxy outgoing requests:
+
+- [custom reverse proxy](https://github.com/vercel/next.js/tree/master/examples/with-custom-reverse-proxy)
+- [redirects](https://nextjs.org/docs/api-reference/next.config.js/redirects)
+
+The problem with the former is that it runs on a different server. When testing local changes on e.g. a phone, we will get unresolvable CORS issues.
+The problem with the latter is that NextJS redirects don't provide a possibility to manipulate requests and/or responses. They just redirect.
+
+## The fix
+
+Prerequisites:
+
+- NextJS [API Routes](https://nextjs.org/docs/api-routes/introduction) with the [Optional catch all API route](https://nextjs.org/docs/api-routes/dynamic-api-routes#optional-catch-all-api-routes)
+- NextJS [API middleware](https://nextjs.org/docs/api-routes/api-middlewares) using [`http-proxy-middleware`](https://github.com/chimurai/http-proxy-middleware)
+- a way to pick the proper URL to the proxy (`PROXY_ENV` in this case)
+
+```js
+// pages/api/[[...slug]].ts
+import { IncomingMessage } from 'http';
+import { createProxyMiddleware } from 'http-proxy-middleware';
+import { NextApiRequest, NextApiResponse } from 'next';
+
+let apiUrl: string;
+
+switch (process.env.PROXY_ENV) { // pick API endpoint depending on the PROXY_ENV, assign to apiUrl }
```

Comment:
This should probably be a multiline comment to avoid commenting out the closing }